cellmlmanip/model.py

```diff
@@ -74,39 +75,59 @@ def __str__(self) -> str:
 class Model(object):
     """An unrolled representation of a CellML model, using list of equations and metadata
     (e.g. units) about symbols used in those equations
+
+    :param name: the name of the model e.g. from <model name="">
     """
-    def __init__(self, name: str) -> None:
-        """Create a new instance of Model object
-        :param name: the name of the model e.g. from <model name="">
-        """
-        self.name: str = name
-        self.units: 'UnitStore' = UnitStore(model=self)
-        self.rdf: rdflib.Graph = rdflib.Graph()
-        self.graph: nx.DiGraph = None
+    def __init__(self, name):
 
-        self.dummy_metadata: Dict[sympy.Dummy, MetaDummy] = OrderedDict()
-        self.name_to_symbol: Dict[str, sympy.Dummy] = dict()
+        self.name = name
+        self.units = UnitStore(model=self)
+        self.rdf = rdflib.Graph()
+        self.graph = None   # An nx.DiGraph
 
-        self.equations: List[sympy.Eq] = list()
+        # Maps sympy.Dummy objects to MetaDummy objects
+        self.dummy_metadata = OrderedDict()
+
+        # Maps string names to sympy.Dummy objects
+        self.name_to_symbol = dict()
+
+        # A list of sympy.Eq objects.
+        self.equations = []
+
+    def add_unit(self, units_name, unit_attributes=None, base_units=False):
+        """
+        Adds information about <units> in <model>
 
-    def add_unit(self, units_name: str, unit_attributes: List[Dict] = None, base_units=False):
-        """Adds information about <units> in <model>
+        :param units_name: A string name
+        :param unit_attributes: An optional list of dictionaries containing ???
+        :base_units: An optional bool that determines ???
         """
         assert not (unit_attributes and base_units), 'Cannot define base unit with unit attributes'
         if base_units:
             self.units.add_base_unit(units_name)
         else:
             self.units.add_custom_unit(units_name, unit_attributes)
 
-    def add_equation(self, equation: sympy.Eq):
-        """Add an equation to this model. Equation must be a Sympy equality"""
+    def add_equation(self, equation):
+        """
+        Adds an equation to this model.
+
+        :param equation: A ``sympy.Eq`` object.
+        """
         assert isinstance(equation, sympy.Eq), 'Equation expression must be equality'
         self.equations.append(equation)
 
-    def add_number(self, *,
-                   number: sympy.Number, units: str, dummy: sympy.Dummy = None) -> sympy.Dummy:
-        """Add metadata about a dummy symbol that represents a number in equations. Returns the
-        sympy.Dummy object used to represent this number"""
+    # TODO: Do we need the * here?
```

Comment:
Yep, says arguments after must be keyword-only. Same below.

Reply:
What's the rationale for that?

Reply:
It's a defensive programming safety feature. Means users can't get the order of arguments wrong, and makes user code more readable.